lib/gringotts/gateways/we_pay.ex

```diff
@@ -0,0 +1,426 @@
+        checkout_id: payment_id
+      })
+
+    commit(:post, "/checkout/capture/", body, opts)
+  end
+
+  @doc """
+  Transfers `amount` from the customer to the merchant.
+
+  wepay attempts to process a purchase on behalf of the customer, by
+  debiting `amount` from the customer's account by charging the customer's
+  `card`.
+
+  ## Example
+  ```
+  iex> amount = Money.new(42, :USD)
+  iex> {:ok, purchase_result} = Gringotts.purchase(Gringotts.Gateways.WePay, amount, card, opts)
+  iex> purchase_result.id # This is the checkout ID
+  ```
+  """
+  @spec purchase(Money.t(), CreditCard.t(), keyword) :: {:ok | :error, Response}
+  def purchase(amount, card = %CreditCard{}, opts) do
+    {currency, value, _} = Money.to_integer(amount)
+
+    with {:ok, card_token_response} <- store(card, opts),
+         {:ok, card_token} <- extract_card_token(card_token_response) do
+      body =
+        Poison.encode!(%{
+          account_id: opts[:account_id],
+          short_description: opts[:short_description],
+          type: opts[:type],
+          amount: value,
+          currency: currency,
+          long_description: opts[:long_description],
+          callback_uri: opts[:callback_uri],
+          auto_release: true,
+          unique_id: opts[:unique_id],
+          reference_id: opts[:reference_id],
+          delivery_type: opts[:delivery_type],
+          payment_method: %{
+            type: "credit_card",
+            credit_card: %{
+              id: card_token
+            }
+          }
+        })
+
+      commit(:post, "/checkout/create/", body, opts)
+    end
+  end
+
+  # purchase with card token.
+  def purchase(amount, card_token, opts) do
+    {currency, value, _} = Money.to_integer(amount)
+
+    body =
+      Poison.encode!(%{
+        account_id: opts[:account_id],
+        short_description: opts[:short_description],
+        type: opts[:type],
+        amount: value,
+        currency: currency,
+        long_description: opts[:long_description],
+        callback_uri: opts[:callback_uri],
+        auto_release: true,
+        unique_id: opts[:unique_id],
+        reference_id: opts[:reference_id],
+        delivery_type: opts[:delivery_type],
+        payment_method: %{
+          type: "credit_card",
+          credit_card: %{
+            id: card_token
+          }
+        }
+      })
+
+    commit(:post, "/checkout/create/", body, opts)
+  end
+
+  @doc """
+  Voids the referenced payment.
+
+  This method attempts a reversal of a previous transaction referenced by
+  `payment_id`.
+
+  > As a consequence, the customer will never see any booking on his statement.
+
+  ## Note
+  > As a consequence, the customer will never see any booking on his statement.
+  > Checkout must be in purchased or captured state.
+
+  ## Example
+  ```
+  iex> {:ok, void_result} = Gringotts.capture(Gringotts.Gateways.WePay, purchase_result.id, opts)
+  ```
+  """
+  @spec void(String.t(), keyword) :: {:ok | :error, Response}
```

Comment:
The `auto_release` setting comes into play here, when we cancel a checkout.
If it was true, the funds are auto-released back to the payer.
otherwise, an explicit call to `checkout/release/` is needed.

Since we don't have any API call for `release`, we must hard-code `auto_release` to true. Hmm... too bad. :disappointed: